src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -757,51 +775,56 @@ bool emberAfOperationalCredentialsClusterUpdateNOCCallback(app::CommandHandler *
 
     FailSafeContext & failSafeContext = DeviceControlServer::DeviceControlSvr().GetFailSafeContext();
 
-    if (!failSafeContext.IsFailSafeArmed(commandObj->GetAccessingFabricIndex()))
-    {
-        LogErrorOnFailure(commandObj->AddStatus(commandPath, Status::UnsupportedAccess));
-        return true;
-    }
-
-    if (failSafeContext.NocCommandHasBeenInvoked())
-    {
-        LogErrorOnFailure(commandObj->AddStatus(commandPath, Status::ConstraintError));
-        return true;
-    }
-
     // Fetch current fabric
     FabricInfo * fabric = RetrieveCurrentFabric(commandObj);
+    fabricIndex         = fabric->GetFabricIndex();
     VerifyOrExit(fabric != nullptr, nocResponse = ConvertToNOCResponseStatus(CHIP_ERROR_INVALID_FABRIC_INDEX));
```

Comment:
Why are we bothering with a null-check after we dereferenced the pointer?

Was this tested with UpdateNOC over PASE before AddNOC (which is when this code would be relevant)?

And per spec, shouldn't that case end up with `InsufficientPrivilege`?  And similar, actually, for PASE after AddNOC....

Reply:
Please move `VerifyOrExit(fabric != nullptr)`

How can fabric be null here though?

Reply:
It will be null if this command came in over PASE before AddNOC.

Reply:
Done